### PR TITLE
Nothingness visits Japan in the winter

### PR DESCRIPTION
**Pull request by alexdowad**
Starting a new batched PR with enhancements to `NodePattern`.

### CODE REVIEW

File: spec/rubocop/node_pattern_spec.rb [hunk 1]
```diff
@@ -86,6 +86,14 @@
...
   end
 
+  describe 'literals' do
+    context 'negative integer literals' do
+      let(:pattern) { '(int -100)' }
+      let(:ruby) { '-100' }
+      it_behaves_like :matching
+    end
+  end
+
```
Comment (DNNX) [inline spec/rubocop/node_pattern_spec.rb]: Is it possible to also add negative float literals? `(float 1.0)` is supported at the moment, but not `(float -1.0)`.

Comment (alexdowad) [inline spec/rubocop/node_pattern_spec.rb]: I was just doing this as you commented.

Comment (DNNX) [inline spec/rubocop/node_pattern_spec.rb]: Awesome, thanks for doing this.

**Comment by alexdowad** [comment]
@DNNX, if you want to use the new pattern features in your open PR, you can base it on this branch. Then after this one is merged, yours will also merge cleanly. Or, if yours is merged first, these commits will go in at the same time.

**Pull request merged**